web/src/stream_create.ts

```diff
@@ -99,10 +100,16 @@ class StreamSubscriptionError {
 const stream_subscription_error = new StreamSubscriptionError();
 
 class StreamNameError {
-    report_already_exists(error?: string): void {
-        const error_message =
-            error ?? $t({defaultMessage: "A channel with this name already exists."});
-        $("#stream_name_error").text(error_message);
+    report_already_exists(rendered_error?: string): void {
+        const error_html =
+            rendered_error ??
+            render_channel_name_conflict_error({
+                stream_id: 0,
+                is_archived: false,
+                show_rename: false,
+                can_view_channel: false,
```

Comment:
Instead of doing this here, I think would be better to just render the error at the line where `report_already_exists` is called without passing the `rendered_error` string. That would make the reason to choose the values for this parameters more clear.

Also, I think we can either skip passing `stream_id` here or pass `undefined` instead of `0`.